File: tooldemo/src/main/java/com/excellence/tooldemo/databinding/CommonBindingAdapterActivity.java
package com.excellence.tooldemo.databinding;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Toast;

import com.excellence.basetoolslibrary.databinding.CommonBindingAdapter;
import com.excellence.tooldemo.BR;
import com.excellence.tooldemo.R;
import com.excellence.tooldemo.bean.databinding.Flower;

import java.util.ArrayList;
import java.util.List;

public class CommonBindingAdapterActivity extends AppCompatActivity
{
	private ActivityCommonBindingAdapterBinding mBinding = null;

	private List<Flower> mFlowers = null;

	@Override
	protected void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);
		mBinding = DataBindingUtil.setContentView(this, R.layout.activity_common_binding_adapter);

		initAdapter();
	}

	private void initAdapter()
	{
		mFlowers = new ArrayList<>();
		for (int i = 0; i < 10; i++)
			mFlowers.add(new Flower("Flower" + i, R.drawable.logo));
		CommonBindingAdapter<Flower> adapter = new CommonBindingAdapter<>(mFlowers, R.layout.item_flower, BR.flower);
		mBinding.setAdapter(adapter);
		mBinding.setOnItemClickListener(new AdapterView.OnItemClickListener()
		{
			@Override
			public void onItemClick(AdapterView<?> parent, View view, int position, long id)
			{
				Toast.makeText(CommonBindingAdapterActivity.this, "点击了" + position, Toast.LENGTH_SHORT).show();
			}
		});
	}

}

File: tooldemo/src/main/java/com/excellence/tooldemo/databinding/CommonBindingAdapterActivity.kt
package com.excellence.tooldemo.databinding

import android.os.Bundle
import android.widget.AdapterView
import android.widget.BaseAdapter
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.excellence.basetoolslibrary.databinding.CommonBindingAdapter
import com.excellence.tooldemo.BR
import com.excellence.tooldemo.R
import com.excellence.tooldemo.bean.databinding.Flower
import java.util.*

class CommonBindingAdapterActivity : AppCompatActivity() {

    private var mBinding: ActivityCommonBindingAdapterBinding? = null
    private var mFlowers: MutableList<Flower> = ArrayList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_common_binding_adapter)
        initAdapter()
    }

    private fun initAdapter() {
        for (i in 0..9) {
            mFlowers.add(Flower("Flower$i", R.drawable.logo))
        }

        val adapter = CommonBindingAdapter(mFlowers, R.layout.item_flower, BR.flower)
        mBinding!!.adapter = adapter
        mBinding!!.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            val flower = mFlowers[position]
            val name = flower.name
            val flag = "[点击]"
            if (name!!.contains(flag)) {
                flower.name = name.replace(flag, "")
            } else {
                flower.name = name + flag
            }
            (parent.adapter as BaseAdapter).notifyDataSetChanged()
            Toast.makeText(this@CommonBindingAdapterActivity, "点击了$position", Toast.LENGTH_SHORT).show()
        }
    }
}